base/gocb_utils.go

```diff
@@ -161,3 +163,36 @@ func getRootCAs(ctx context.Context, caCertPath string) (*x509.CertPool, error)
 	}
 	return rootCAs, nil
 }
+
+func GetServerUUIDWithAgent(a *gocbcore.Agent, mgmtEp, username, password string) (string, error) {
```

Comment:
I'd probably create a common function name with `MgmtRequest` with this call string and have `GocbV2Bucket.MgmtRequest` call into this.

Then you can replace reuse this in https://github.com/couchbase/sync_gateway/blob/b77c0b493aea1f619023a2f3ea03945f015dc177/base/bucket.go#L375.